### PR TITLE
cli: remove deprecated try-runtime subcommand (also companion for #14731)

### PR DESCRIPTION
**Pull request by liamaharon**
Companion for https://github.com/paritytech/substrate/pull/14731

try-runtime subcommand is being deprecated.

This PR replaces the subcommand with a stub/deprecation notice which includes instructions on how to use the standalone CLI.

### TODO before merge
- [x] Remove `--branch liam-v0.1.0` from CI command once https://github.com/paritytech/try-runtime-cli/pull/6 is merged.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3363393

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3365165

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3365426

File: scripts/ci/gitlab/pipeline/check.yml [hunk 1]
```diff
@@ -62,10 +62,10 @@ check-try-runtime:
     - |
       export RUST_LOG=remote-ext=debug,runtime=debug
       echo "---------- Running try-runtime for ${NETWORK} ----------"
-      time cargo build --release --locked -p "$NETWORK"-runtime
-      time cargo run --locked --release --features try-runtime try-runtime \
+      time cargo install --git https://github.com/paritytech/try-runtime-cli --branch liam-v0.1.0
```
Comment (ggwpez) [inline scripts/ci/gitlab/pipeline/check.yml:65]: Why do we need to use this special branch here?

Comment (liamaharon) [reply scripts/ci/gitlab/pipeline/check.yml:65]: the branch hasn't merged yet. I have a TODO in the PR description to remove the `--branch` before we merge

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3396522

**Review by sergejparity — APPROVED** [review]

File: scripts/ci/gitlab/pipeline/check.yml [hunk 2]
```diff
@@ -62,10 +62,10 @@ check-try-runtime:
     - |
       export RUST_LOG=remote-ext=debug,runtime=debug
       echo "---------- Running try-runtime for ${NETWORK} ----------"
-      time cargo build --release --locked -p "$NETWORK"-runtime
-      time cargo run --locked --release --features try-runtime try-runtime \
+      time cargo install --git https://github.com/paritytech/try-runtime-cli
```
Comment (alvicsam) [inline scripts/ci/gitlab/pipeline/check.yml:65]: Is it possible to pin the version with tag or sha?

**Review by alvicsam — CHANGES_REQUESTED** [review]

**Review by alvicsam — APPROVED** [review]

**Review by EgorPopelyaev — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3397626

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3398428

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/3402406

**Review by sam0x17 — APPROVED** [review]

**Pull request merged**